src/main/java/com/boardgamesworld/bgrental/boardgame/infrastructure/api/BoardGameController.java

```diff
@@ -0,0 +1,124 @@
+package com.boardgamesworld.bgrental.boardgame.infrastructure.api;
+
+import com.boardgamesworld.bgrental.boardgame.domain.BoardGame;
+import com.boardgamesworld.bgrental.boardgame.domain.BoardGameFacade;
+import com.boardgamesworld.bgrental.boardgame.domain.InvalidBoardGameException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/boardgames")
+public class BoardGameController {
+
+    private BoardGameFacade boardGameFacade;
+    private Logger logger = LoggerFactory.getLogger(BoardGameController.class);
+
+    public BoardGameController(BoardGameFacade boardGameFacade) {
+        this.boardGameFacade = boardGameFacade;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<BoardGameDto>> getAllBoardGames() {
+        List<BoardGame> boardGames = boardGameFacade.getAllBoardGames();
+        List<BoardGameDto> boardGamesDto;
+
+        if (boardGames != null) {
+            boardGamesDto = boardGames.stream()
+                    .map(boardGame -> new BoardGameDto(
+                            boardGame.getBoardGameId(),
+                            boardGame.getName(),
+                            boardGame.getPricePerDay(),
+                            boardGame.isRented(),
+                            boardGame.getCondition(),
+                            boardGame.getDetails()
+                    ))
+                    .collect(Collectors.toList());
+
+            return ResponseEntity.ok().body(boardGamesDto);
+        } else {
+            return ResponseEntity.notFound().build();
```

Comment:
`GET /boardgames` nie powinien zwrócić 404, a pustą listę.

Reply:
404 to błąd klienta, a jaki jest jego tutaj błąd? Pyta o gry, a jak nie ma żadnej to dostaje pustą listę. 

Jeżeli pyta o szczegóły konkretnej gry, np. GET /boardgames/ID to wtedy 404 oznacza, że taki gra (zasób) nie istnieje